fish.py

```diff
@@ -1285,7 +1306,11 @@ def fish_list_keys(buffer):
 
     for (target, key) in sorted(fish_keys.items()):
         (server, nick) = target.split("/")
-        weechat.prnt(buffer, "\t%s(%s): %s" % (nick, server, key))
+        if filterOnServer:
+            if server == filterOnServer:
+                weechat.prnt(buffer, "\t%s(%s): %s" % (nick, server, key))
+        else:
+            weechat.prnt(buffer, "\t%s(%s): %s" % (nick, server, key))
```

Comment:
maybe it would be nicer to avoid the duplication here
```suggestion
        if filterOnServer and server != filterOnServer:
            continue
        weechat.prnt(buffer, "\t%s(%s): %s" % (nick, server, key))
```